src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -286,26 +286,18 @@ internal void ExtractRelativeToDirectory(string destinationDirectoryPath, bool o
                 Directory.CreateDirectory(Path.GetDirectoryName(fileDestinationPath)!);
                 ExtractToFileInternal(fileDestinationPath, linkTargetPath, overwrite);
             }
+        }
 
-            // If the path can be extracted in the specified destination directory, returns the full path with sanitized file name. Otherwise, throws.
-            static string GetSanitizedFullPath(string destinationDirectoryFullPath, string path, string exceptionMessage)
-            {
-                string actualPath = Path.Join(Path.GetDirectoryName(path), ArchivingUtils.SanitizeEntryFilePath(Path.GetFileName(path)));
-
-                if (!Path.IsPathFullyQualified(actualPath))
-                {
-                    actualPath = Path.Combine(destinationDirectoryFullPath, actualPath);
-                }
-
-                actualPath = Path.GetFullPath(actualPath);
-
-                if (!actualPath.StartsWith(destinationDirectoryFullPath, PathInternal.StringComparison))
-                {
-                    throw new IOException(string.Format(exceptionMessage, path, destinationDirectoryFullPath));
-                }
+        // If the path can be extracted in the specified destination directory, returns the full path with sanitized file name. Otherwise, throws.
+        private static string GetSanitizedFullPath(string destinationDirectoryFullPath, string path, string exceptionMessage)
```

Comment:
Fetching exception message from resources is relatively expensive. It should be done only if we are actually going to throw an exception.

It would be better for this method to return null and let caller throw the exception with the desired message.

Reply:
Didn't know, thanks for pointing it out. Will change it.